### ISSUE TITLE
com.redhat.lightblue.eval.Updater is unable to handle non-json backends

### ISSUE THREAD

**Issue by dcrissman** [lightblue-platform/lightblue-core#673]
The core logic to process UpdateExpression objects is tightly coupled to updating json documents. This needs to be split so that non-json backends (eg. ldap, rdbms) can reuse the the processing logic, but then create updates in a way that is appropriate for the target backend. I am thinking something like QueryIteratorSkeleton.

**Comment by bserdar** [comment]
This is not easy. Lets do this after we have another backend, ldap or
rdbms.

**Comment by dcrissman** [comment]
Agreed that this is a non-trivial change. I actually found this while looking into ldap update [1] functionality.

[1] https://github.com/lightblue-platform/lightblue-ldap/issues/8
